indra/newview/scripts/lua/LLChatListener.lua

```diff
@@ -0,0 +1,41 @@
+local fiber = require 'fiber'
+local inspect = require 'inspect'
+
+local LLChatListener = {}
+local waitfor = {}
+
+function LLChatListener:new()
+    local obj = setmetatable({}, self)
+    self.__index = self
+    obj.name = 'Chat_listener'
+
+    return obj
+end
+
+function LLChatListener:handleMessages(event_data)
+    --print(inspect(event_data))
+    return true
+end
+
+function LLChatListener:start()
+    waitfor = leap.WaitFor:new(-1, self.name)
+    function waitfor:filter(pump, data)
+        return data
```

Comment:
Using `LLLeapListener`'s "listen" operation, `pump` should arrive as `"LLNearbyChat"`. Your `filter()` function should return only those -- otherwise your chat listener will echo all incoming events, e.g. user floater clicks, ...